src/posit/connect/external/databricks.py

```diff
@@ -0,0 +1,66 @@
+import abc
+import os
+from typing import Callable, Dict, Optional
+
+from ..client import Client
+from ..oauth import OAuthIntegration
+
+HeaderFactory = Callable[[], Dict[str, str]]
+
+# https://github.com/databricks/databricks-sdk-py/blob/v0.20.0/databricks/sdk/credentials_provider.py
+# https://github.com/databricks/databricks-sql-python/blob/v3.1.0/src/databricks/sql/auth/authenticators.py
+# In order to keep compatibility with the Databricks SDK
+class CredentialsProvider(abc.ABC):
+    """CredentialsProvider is the protocol (call-side interface)
+     for authenticating requests to Databricks REST APIs"""
+
+    @abc.abstractmethod
+    def auth_type(self) -> str:
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def __call__(self, *args, **kwargs) -> HeaderFactory:
+        raise NotImplementedError
+
+
+class PositOAuthIntegrationCredentialsProvider(CredentialsProvider):
+    def __init__(self, posit_oauth: OAuthIntegration, user_identity: str):
+        self.posit_oauth = posit_oauth
+        self.user_identity = user_identity
+
+    def auth_type(self) -> str:
+        return "posit-oauth-integration"
+
+    def __call__(self, *args, **kwargs) -> HeaderFactory:
+        def inner() -> Dict[str, str]:
+            access_token = self.posit_oauth.get_credentials(self.user_identity)['access_token']
+            return {"Authorization": f"Bearer {access_token}"}
+        return inner
+
+
+# Use these environment varariables to determine if the
+# client SDK was initialized from a piece of content running on a Connect server.
+def is_local() -> bool:
+    return not os.getenv("CONNECT_SERVER") and not os.getenv("CONNECT_CONTENT_GUID")
```

Comment:
I don't think `CONNECT_SERVER` is right here: that's what we recommend folks set locally in all of our API examples (e.g. https://docs.posit.co/connect/cookbook/). 

I'd rather use the accurate variable even if it says `RSTUDIO`; we can fix that on the server going forward and (eventually) purge it here.